gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/swagger/GraviteeApiDefinition.java

```diff
@@ -15,42 +15,61 @@
  */
 package io.gravitee.rest.api.management.rest.resource.swagger;
 
-import io.swagger.annotations.SwaggerDefinition;
-import io.swagger.jaxrs.Reader;
-import io.swagger.jaxrs.config.ReaderListener;
-import io.swagger.models.Swagger;
-import io.swagger.models.Tag;
-import io.swagger.models.auth.BasicAuthDefinition;
-import java.util.Comparator;
-import java.util.TreeMap;
+import io.gravitee.common.util.Version;
+import io.swagger.v3.jaxrs2.ReaderListener;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.integration.api.OpenApiReader;
+import io.swagger.v3.oas.models.*;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import io.swagger.v3.oas.models.servers.Server;
+import io.swagger.v3.oas.models.tags.Tag;
+import java.util.*;
 
 /**
  * @author David BRASSELY (david.brassely at graviteesource.com)
  * @author GraviteeSource Team
  */
-@SwaggerDefinition
+@OpenAPIDefinition
 public class GraviteeApiDefinition implements ReaderListener {
 
     public static final String TOKEN_AUTH_SCHEME = "gravitee-auth";
 
     @Override
-    public void beforeScan(Reader reader, Swagger swagger) {}
+    public void beforeScan(OpenApiReader reader, OpenAPI openAPI) {
+        openAPI
+            .info(new Info().version(Version.RUNTIME_VERSION.MAJOR_VERSION).title("Gravitee.io - Management API"))
+            .servers(Collections.singletonList(new Server().url("http://localhost:8083/management")));
```

Comment:
This seems kind of tricky, since the `GraviteeApiDefinition` is not a bean and we have no access to the bean factory. We can make some dirty casts and reflections to get the information... :-(

Maybe you have any better idea?

Reply:
I must admit I have no idea for the moment. When I have time, I will think about it.
@brasseld maybe a 💡 ?

Reply:
I've found a way to get the name dynamically

Reply:
I've found a way to get the name dynamically

Reply:
I've found a way to get the name dynamically

Reply:
I've found a way to get the name dynamically

Reply:
I've found a way to get the name dynamically

Reply:
Awesome ! I think you should also take `jetty.secured`, `jetty.host` and `jetty.port` to have a full URL
https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-container/src/main/java/io/gravitee/rest/api/standalone/jetty/JettyConfiguration.java

Reply:
>  think you should also take `jetty.secured`, `jetty.host` and `jetty.port` to have a full URL

This will not work for cloud deployments and also in a default dev setup `https://0.0.0.0:8083` is not what  the URL should look like.

I'm thinking of using (in this order):

1. `console.api`
    * which is not set by default / via helm
    * we should ensure this is set via helm charts similar to https://github.com/gravitee-io/helm-charts/blob/f337a4948cbb2ed3b3281739f530c57b84b44ecb/apim/3.x/templates/ui/ui-deployment.yaml#L87-L88
2. Fallback to jetty settings replacing `0.0.0.0` with `localhost`
    * In `GraviteeOpenApiReader` we have no access to the class `JettyConfiguration.java`, so we need to reolve it with spring magic

wdyt?

Reply:
I agree with you.
I think you forgot `.url` in `console.api.url`

I agree for the helm-charts, we could define console.ui.url & console.api.url the value in https://github.com/gravitee-io/helm-charts/blob/master/apim/3.x/templates/api/api-configmap.yaml

Reply:
https://github.com/gravitee-io/helm-charts/pull/220